core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexWaitsIT.java

```diff
@@ -120,10 +122,66 @@ public void testCreateIndexWaitsForAllActiveShards() throws Exception {
+                                .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(1, 5))
+                                .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), internalCluster().numDataNodes() - 1)
+                                .build();
+
+        logger.info("--> start the index creation process");
+        ListenableActionFuture<CreateIndexResponse> responseListener =
+            prepareCreate(indexName)
+                .setSettings(settings)
+                .setWaitForActiveShards(ActiveShardCount.ALL)
+                .execute();
+
+        logger.info("--> wait until the cluster state contains the new index");
+        assertBusy(() -> assertTrue(client().admin().cluster().prepareState().get().getState().metaData().hasIndex(indexName)));
+
+        logger.info("--> delete the index");
+        assertAcked(client().admin().indices().prepareDelete(indexName));
+
+        logger.info("--> ensure the create index request completes");
+        CreateIndexResponse response = responseListener.get();
+        assertAcked(response);
+        assertFalse(response.isTimedOutWaitingForShards());
+    }
+
+    public void testIndexWaitSettingUpdated() throws Exception {
+        final String indexName = "test-idx";
+
+        // updating wait_for_active_shards to an invalid number should fail
+        int numReplicas = internalCluster().numDataNodes() - 1;
+        final Settings settings = Settings.builder()
+                                      .put(indexSettings())
+                                      .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(1, 5))
+                                      .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), numReplicas)
+                                      .put(WAIT_FOR_ACTIVE_SHARDS_SETTING.getKey(),
+                                           numReplicas + randomIntBetween(2, 5)) // exceeds # of shards
+                                      .build();
+        expectThrows(IllegalArgumentException.class, () -> prepareCreate(indexName).setSettings(settings).get());
+
+        // update wait_for_active_shards to a valid value, should honor it, but timeout as there aren't enough data nodes
+        numReplicas = internalCluster().numDataNodes() + randomInt(4);
+        final Settings settings2 = Settings.builder()
+                                       .put(indexSettings())
+                                       .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(1, 5))
+                                       .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), numReplicas)
+                                       .put(WAIT_FOR_ACTIVE_SHARDS_SETTING.getKey(), "all")
+                                       .build();
+        assertTrue(prepareCreate(indexName)
+                       .setSettings(settings2)
+                       .setTimeout("100ms")
+                       .get()
+                       .isTimedOutWaitingForShards());
+    }
```

Comment:
how about extending this test in the following way:

create index with WAIT_FOR_ACTIVE_SHARDS_SETTING higher than number of data nodes and timeout 1h.
check that future is not resolved
update index settings WAIT_FOR_ACTIVE_SHARDS_SETTING to number lower than data nodes.
check that previous future is now resolved (future.get(10, seconds)).